modules/cluster-management/src/main/java/org/apache/ignite/internal/cluster/management/ClusterManagementGroupManager.java

```diff
@@ -350,6 +361,49 @@ public CompletableFuture<Void> startAsync(ComponentContext componentContext) {
         return nullCompletedFuture();
     }
 
+    private CompletableFuture<Void> doClusterReset(ResetClusterMessage resetClusterMessage) {
+        LOG.info("Found a ResetClusterMessage in storage, going to do cluster reset [message={}]", resetClusterMessage);
+
+        return destroyCmgWithEvents()
+                .thenCompose(unused -> {
+                    if (resetClusterMessage.cmgNodes().contains(clusterService.nodeName())) {
```

Comment:
Is it possible for multiple nodes to start and have two different reset messages in their storages?

Reply:
And what will happen? The first received init will be used and other rejected, so we will eventually end up with two disjoint clusters?

Reply:
If the user will send the same cmgNodes twice, once to one conductor, another time to another, then only one of them will form a majority; another 'cluster' will not have a functioning CMG. These nodes will be picked up using 'migrate' later.

If the user uses 2 disjoint sets of nodes as new CMGs, they may get 2 disjoint clusters. This is something that we cannot prevent. In following phases of this design this could be addressed (we could try to detect such situations and force nodes to stop).

Reply:
thank you for the explanation